include/ur_client_library/comm/producer.h

```diff
@@ -105,24 +108,24 @@ class URProducer : public IProducer<T>
     // 4KB should be enough to hold any packet received from UR
     uint8_t buf[4096];
     size_t read = 0;
-    // expoential backoff reconnects
+    // exponential backoff reconnects
     while (true)
     {
+      if (!running_)
+        return true;
+
       if (stream_.read(buf, sizeof(buf), read))
       {
         // reset sleep amount
-        timeout_ = std::chrono::seconds(1);
+        timeout_ = std::chrono::milliseconds(100);
         BinParser bp(buf, read);
         return parser_.parse(bp, products);
       }
 
-      if (!running_)
-        return true;
-
```

Comment:
Why exactly did you drop this? I don't remember out of my head why we have it there, but I would be hesitant deleting this without further information.